File: Lista1/Craps.java
import java.util.Random;

class Craps
{
    public static void main(String[] args)
    {
        Random dados = new Random();

        int d1, d2,i=1,a=0;
        d1 = dados.nextInt(6)+1;
        d2 = dados.nextInt(6)+1;

        System.out.println("primeiro dado:" + d1 + "\tsegundo dado:" + d2 + "\n");
        System.out.println("soma dos dados:" + (d1+d2));
        
        if(d1+d2==7 || d1+d2==11)
        {
            System.out.println("voce ganhou!");
        }
        else if(d1+d2==2 || d1+d2==3 || d1+d2==12)
        {
            System.out.println("voce perdeu!");
        }
        else
        {
            System.out.println("ponto:" + (d1+d2));
            a=d1+d2;
             System.out.println("Estagio 2");
             while(true)
            {
                d1 = dados.nextInt(6)+1;
                d2 = dados.nextInt(6)+1;
                System.out.println("dado1: " + d1 + " dado2:" + d2);
                System.out.println("jogada " + i + ": " + (d1+d2));
                i++;
                if(d1+d2==7)
                {
                    System.out.println("voce perdeu!");
                    break;
                }
                else if((d1+d2)==a)
                {
                    System.out.println("voce ganhou!");
                    break;
                }
            }
        }
    }
}

File: Lista2/craps.py
from random import randint

dado1 = randint(1,7)
dado2 = randint(1,7)
ponto=dado1+dado2

print(f'estagio 1:\ndado 1: {dado1} dado 2: {dado2}\nponto: {ponto}' )
v= [2,3,12]
if ponto== 7 or ponto==11:
    print('voce ganhou')

elif ponto in v:
    print('voce perdeu')

else:
    print('estagio 2:')

while True:
    dado1 = randint(1,7)
    dado2 = randint(1,7)
    print(f'dado1: {dado1} dado 2:{dado2}\nsoma:{dado1+dado2}')
    if dado1+dado2==7:
        print('voce perdeu')
        break
    elif dado1+dado2==ponto:
        print('parabens,voce ganhou')
        break
